components/layout/fragment.rs

```diff
@@ -1976,6 +1974,11 @@ impl Fragment {
         if self.style().get_effects().transform.is_some() {
             return true
         }
+        // Canvas always layerizes, as an special case
```

Comment:
Could you add something like "FIXME(pcwalton): Don't unconditionally form stacking contexts for each canvas." here? @rocallahan wants us to fix this at some point and I agree with him. :) I'd be happy to elaborate more on the long-term plans if you're interested.